src/S3Control/S3ControlClient.php

```diff
@@ -211,9 +215,55 @@ public function __construct(array $args)
                         ? $args['endpoint']
                         : null,
                     'use_fips_endpoint' => $this->getConfig('use_fips_endpoint'),
-                ]
+                ],
+                $this->isUseEndpointV2()
             ),
             's3control.endpoint_arn_middleware'
         );
     }
+
+    /**
+     * Modifies API definition to remove `AccountId`
+     * host prefix.  This is now handled by the endpoint ruleset.
+     *
+     * @return void
+     *
+     * @internal
+     */
+    private function processEndpointV2Model()
+    {
+        $definition = $this->getApi()->getDefinition();
+        $this->removeHostPrefix($definition);
+        $this->removeRequiredMember($definition);
+        $this->getApi()->setDefinition($definition);
+    }
+
+    private function removeHostPrefix(&$definition)
+    {
+        foreach($definition['operations'] as &$operation) {
+            if (isset($operation['endpoint']['hostPrefix'])
+                && $operation['endpoint']['hostPrefix'] === '{AccountId}.'
+            ) {
+                $operation['endpoint']['hostPrefix'] = str_replace(
+                    '{AccountId}.',
+                    '',
+                    $operation['endpoint']['hostPrefix']
+                );
+            }
+        }
+    }
+
+    private function removeRequiredMember(&$definition)
+    {
+        foreach($definition['shapes'] as &$shape) {
+            if (isset($shape['required'])
+            ) {
+                $found = array_search('AccountId', $shape['required']);
```

Comment:
Why are we doing this?

Reply:
AccountId is technically no longer required (can be parsed out of an ARN), but the model will not be changed. Ruby and Python have had to make similar changes.